lib/matplotlib/font_manager.py

```diff
@@ -855,7 +847,6 @@ def set_weight(self, weight):
         except ValueError:
             if weight not in weight_dict:
                 raise ValueError("weight is invalid")
-            weight = weight_dict[weight]
```

Comment:
Numeric value of `self._weight` could be inaccurate.
For example, FiraSans-Hair and FiraSans-Regular has the same numeric weight (400), and thus they have the same priority for the previous score_weight implementation.
If we store the weight name of a font instead of the possibly inaccurate numeric weight, we are able to score the font more accurately.